crates/core_simd/examples/dot_product.rs

```diff
@@ -0,0 +1,31 @@
+// Code taken from the `packed_simd` crate
+// Run this code with `cargo test --example dot_product`
+#![feature(array_chunks)]
+use core_simd::*;
+
+pub fn dot_prod(a: &[f32], b: &[f32]) -> f32 {
+    assert_eq!(a.len(), b.len());
+
+    // TODO handle remainder when a.len() % 4 != 0
+    a.array_chunks::<4>()
+        .map(|&a| f32x4::from_array(a))
+        .zip(b.array_chunks::<4>().map(|&b| f32x4::from_array(b)))
+        .map(|(a, b)| (a * b).horizontal_sum())
```

Comment:
This is a pretty bad dot-product example, honestly. You want to use something closer to the algorithm [here](https://gist.github.com/thomcc/4aceff7f4482dd6ab1eabe391e843c49#file-dot-bench-rs-L256-L265) to defer the hsum until the end, as opposed to doing one hsum per chunk (which is going to be terrible for perf).

That said, I guess it's just ported from packed_simd, and we can probably improve it later...

Reply:
Oh, this was already brought up.